lib/core/Canvas.js

```diff
@@ -221,18 +224,34 @@ Canvas.prototype.getDefaultLayer = function() {
  *
  * @returns {Snap<SVGGroup>}
  */
-Canvas.prototype.getLayer = function(name) {
+Canvas.prototype.getLayer = function(name, index) {
 
   if (!name) {
     throw new Error('must specify a name');
   }
 
   var layer = this._layers[name];
+
   if (!layer) {
-    layer = this._layers[name] = createGroup(this._viewport, 'layer-' + name);
+    if (!index) {
+      index = 0;
+    }
+
+    var childIndex = 0;
+
+    mapValues(this._layers, function(layer) {
```

Comment:
`this._layers` is now an object since the index of each layer must be remembered. Reduce could therefore only be used with lodash's `values` or `Object.values` which is not supported by IE. For that reason I chose `mapValues`. Any other ideas?